### PR TITLE
Our Engineering Experience with React and Storybook

### PR DESCRIPTION
**Pull request by dan-auth0**
❗️ ~Do not merge until https://github.com/auth0/blog/pull/1661 is merged, please. This post relies on that new `include`.~

PR #1661 has been merged. Ready to post.

### CODE REVIEW

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  - ux
+  - engineering
+  - component-library
+  - reusability
+  - banking
+related:
+  - 
+---
+
+At Auth0, we want our customers to be successful using our [authentication solutions](https://auth0.com/b2c-customer-identity-management). The Customer Success Infrastructure Team builds customer-facing and internal tools that help us achieve that goal. One of the technologies that have enabled them to have rapid iterations in user interface development is [React](https://reactjs.org/).
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: > ... help us achieve that goal.

wouldn't "this" be more appropriated as you just mentioned the goal?

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+<a href="https://auth0.com/authenticate/react/" style="text-align: center;" title="Authenticate React using Auth0">
+  <img src="https://cdn.auth0.com/blog/design-samples/react-auth0-banner.png" alt="Authenticate React using Auth0">
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: this is an ugly image

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+</a>
+
+This team builds a variety of different projects such as the software that runs our [Auth0 Support Center](https://support.auth0.com/) and other internal tools that power internal support workflows. In addition to React, they develop their projects using [Node.js](https://nodejs.org/en/), [hapi.js](https://hapijs.com/), and [Redux](https://redux.js.org/). It is a JavaScript-heavy team and that is driven by building quality features quickly.
+
+<p style="text-align: center;">
+  <img src="https://cdn.auth0.com/blog/storybook-intro/auth0-customer-success-team.png" alt="Auth0 Customer Success Infrastructure Team">
+</p>
+
+> Would you like to be part of a Team like this? We are currently hiring engineers to [join the Customer Success Infrastructure Team](https://auth0.com/careers/job/engineer-customer-success-tools:28144879-ee27-41bc-ae30-fd3cba1f309a)!
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: Team is not a proper noun. So, no capital T needed.

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+In this blog post, through Guillermo's insight, we are going to explore how the Account Center Team is using [React and Storybook](https://storybook.js.org/basics/guide-react/) to optimize UI development and promote consistent branding by creating a component library that can be shared across different projects.
+
+## Don't Repeat Yourself (DRY)
+
+[Auth0 Support Center](https://support.auth0.com/), a customer-facing site, and Customer Success Tools, an internal site, have similar elements on their interfaces. For example, despite having a different business context in each interface, the _Quota Utilization_ feature in both interfaces uses the same structure to report data to the user.
+
+<p style="text-align: center;">
+  <img src="https://cdn.auth0.com/blog/auth-storybook/support-center-vs-cs-tools-component.png" alt="Two identical React component defined in different codecases">
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: codecases? is this a thing?

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+
+For example, when creating a component, we started asking the following questions:
+
+- How to categorize the component within a library? Should we have categories at all?
+- How should a prototype of the component be presented? Should a blank page with mock data be used?
+- Should different states of the component be presented on its prototype?
+
+It was critical to be able to answer these questions visually and in an organized manner to help us iterate cleanly through our feature development process. This process typically consists of:
+
+- Creating an MVP (Minimum Viable Product) or POC (Proof of Concept)
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: the other items end with a full stop (although they are not really complete sentences...)

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+In React, there are different aspects of UI Testing. We categorize them as follows along with their tooling:
+
+- Structural Testing:
+  - [Enzyme](https://github.com/airbnb/enzyme)
+  - [Jest Snapshot Testing](https://jestjs.io/docs/en/snapshot-testing)
+- Interaction Testing
+  - Enzyme
+- CSS/Styles Testing:
+
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: you probably noticed already, but whenever I see blank lines, I wonder if they don't break jekyll... anyway, I suppose you tested it locally

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  - Image comparison to do regression testing
+  - Jest Snapshot using inline styles
+  - Other tools like [BackstopJS](https://garris.github.io/BackstopJS/), [PhantomCSS](https://github.com/HuddleEng/PhantomCSS), [Gemini](https://gemini-testing.github.io/), and [Happo](https://github.com/Galooshi/happo)
+
+- Manual Testing
+
+Storybook integrates with Jest Snapshot through an add-on called [`StoryShots`](https://github.com/storybooks/storybook/tree/master/addons/storyshots/storyshots-core). `StoryShots` adds automatic Jest Snapshot Testing to our codebase by using our existing Storybook stories as the input for Jest Snapshot Testing.
+
+With our Heroku CI pipeline configured, each time someone makes a PR with a new component against our Storybook, this automated testing is triggered and a visual preview of our app is created. This lets us perform structural and style testing much faster.
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: I noticed that you mentioned PR at least twice in this article. The problem is that there is no definition of what PR stands for...

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+## Sharing our Engineering Knowledge
+
+We are happy to share what our engineering process looks like in detail and how it has positively impacted our developer experience in creating, testing, and documenting user interfaces. Collaborating directly with Guillermo, we are going to explore how to use Storybook with React, our Storybook-Driven Development approach, the optimizations we made to testing React components and the birth of the Design Systems Team that merges design and development effectively.
+
+Please stay tuned for more on this React at Auth0 series! You may stay in touch by following [@auth0 on Twitter](https://twitter.com/auth0), subscribing to our newsletter, or checking this blog back from time to time. We hope that you'll love the content that is coming up.
+
+{% include asides/about-auth0.markdown %}
```
Comment (brunokrebs) [inline _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: wouldn't it be better to add the React aside?

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown (hunk 1 above)
Comment (dan-auth0) [reply _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: `that` is fine as it refers to a previous statement.

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown (hunk 2 above)
Comment (dan-auth0) [reply _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: That was done by design :)

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown (hunk 4 above)
Comment (dan-auth0) [reply _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: Typo. Fixed.

File: _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown (hunk 8 above)
Comment (dan-auth0) [reply _posts/2018-08-20-our-engineering-experience-with-react-and-storybook.markdown]: It's too long and doesn't compliment the theme of the post.

**Pull request merged**